src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -498,9 +500,44 @@ private void ProcessRenderQueue()
             {
                 ProcessRenderQueue();
             }
+
+            if (asyncDisposalTasks != null)
+            {
+                _ = ProcessAsyncDisposables(asyncDisposalTasks);
+            }
+
             ComponentsProfiling.Instance.End();
         }
 
+        private async Task ProcessAsyncDisposables(List<Task> asyncDisposalTasks)
+        {
+            await Task.WhenAll(asyncDisposalTasks);
```

Comment:
Won't Task.WhenAll raise an exception if a disposal task throws?

Reply:
Yes that does seem like an issue here

Reply:
I got rid of all this